### ISSUE TITLE
Issue with separated_list and &[u8]

### ISSUE THREAD

**Issue by resistor** [Geal/nom#717]
## Prerequisites

- Rust version : rustc 1.24.1 (d3ae9a9e0 2018-02-27)
- nom version : 4.0.0-beta2
- nom compilation features used: defaults

## Test case

```
#[macro_use]
extern crate nom;

named!(device_tree_strings<&[u8], Vec<&[u8]> >,
    separated_list!(tag!([0x0]), is_not!([0x0]))
);
```

## Error:

```
no method named `find_token` found for type `[{integer}; 1]` in the current scope
```

**Comment by resistor** [comment]
Actually, this doesn't appear to be specific to separated_list.  I'm getting the same error with:

```
named!(device_tree_strings<&[u8], Vec<&[u8]> >,
    many0!(
        do_parse!(
            string: is_not!([0x0]) >>
            tag!([0x0]) >>
            (string)
        )
    )
);
```

**Comment by resistor** [comment]
Interestingly, this one does work:

```
named!(device_tree_strings<&[u8], Vec<&[u8]> >,
    separated_list!(tag!([0x0]), take_while!(|x| x != 0))
);
```

**Comment by Geal** [comment]
Hello, this should be fixed in fd490f6
